libs/driver-adapters/executor/src/qc-test-worker/worker.ts

```diff
@@ -98,25 +98,46 @@ if (!parentPort) {
 
 setupDefaultPanicHandler()
 
+type RpcResponse =
+  | null
+  | { type: "Result"; args: object }
+  | { type: "Error"; args: RpcError };
+
+type RpcError = {
+  message: string
+  code?: string
+  stack?: string
+}
+
 parentPort.on('message', async (rawMsg: unknown) => {
   const msg = S.decodeUnknownSync(Message)(rawMsg)
-  let response: unknown
+  let response: RpcResponse
 
   debug('worker received message:', msg.type)
 
   try {
-    response = await dispatchMessage(msg)
-  } catch (error) {
-    // TODO: we should have a nicer mapping for driver adapter errors
-    response = error instanceof Error ? error : new Error(JSON.stringify(error))
-  }
-
-  // The Rust side expects `TransactionEndResponse::Ok(Empty)`,
-  // where `Empty` is `struct Empty {}` as an empty response.
-  // Without this conversion the test cases don't receive the
-  // response at all, rendering them frozen.
-  if (response === undefined) {
-    response = {}
+    // The Rust side expects `TransactionEndResponse::Ok(Empty)`,
+    // where `Empty` is `struct Empty {}` as an empty response.
+    // Without this conversion the test cases don't receive the
+    // response at all, rendering them frozen.
+    let result = await dispatchMessage(msg)
+    response = result === undefined || result === null
+      ? null
+      : {
+        type: 'Result',
+        args: result!,
+      }
```

Comment:
Same as previous comment, I'd expect us to return something like `{ type: 'Result, args: null }`